core/src/main/scala/org/scalajs/macrotaskexecutor/MacrotaskExecutor.scala

```diff
@@ -152,6 +153,13 @@ object MacrotaskExecutor extends ExecutionContextExecutor {
         // we don't try to look for process.nextTick since scalajs doesn't support old node
         // we're also not going to bother fast-pathing for IE6; just fall through
 
+        if (LinkingInfo.developmentMode) {
+          if (js.typeOf(js.Dynamic.global.console) != Undefined && js.typeOf(js.Dynamic.global.console.warn) != Undefined)
+            js.Dynamic.global.console.warn(
+              "Unable to polyfill setImmediate() in this environment so falling back to setTimeout(); this may affect performance. " +
+                "See https://github.com/scala-js/scala-js-macrotask-executor for more information.")
+        }
```

Comment:
Trying to balance being helpful/informative without being too alarmist, since this is still a _correct_ implementation.